lib/jxl/enc_frame.cc

```diff
@@ -1339,9 +1339,20 @@ Status EncodeFrame(const CompressParams& cparams_orig,
     std::vector<coeff_order_t> ac_group_order(num_groups);
     std::iota(ac_group_order.begin(), ac_group_order.end(), 0);
     size_t group_dim = frame_dim.group_dim;
-    // The center of the image.
-    int64_t imag_cx = ib.xsize() / 2;
-    int64_t imag_cy = ib.ysize() / 2;
+    // The center of the image is either given by parameters or chosen
+    // to be the middle of the image by default if center_x, center_y resp.
+    // are not provided.
+    if (cparams.center_x != size_t(-1)) {
```

Comment:
`static_cast<size_t>(-1)`

Reply:
Done. Should I also cast it when setting the defaults in enc_params.h?